src/ui/map.js

```diff
@@ -2892,7 +2922,15 @@ class Map extends Camera {
         // A task queue callback may have fired a user event which may have removed the map
         if (this._removed) return;
 
-        this._updateProjection();
+        if (this.getProjection().name === 'globe') {
+            if (this.transform.zoom >= GLOBE_ZOOM_THRESHOLD_MAX) {
+                if (this._runtimeProjection && this._runtimeProjection.name === 'globe') {
+                    this._updateProjection();
+                }
+            } else if (!this._runtimeProjection || this._runtimeProjection.name === 'mercator') {
+                this._updateProjection();
+            }
+        }
```

Comment:
These don't belong in `_updateProjection`, since zoom transitions have no bearing during calls to setProjection().

The check here ensures that projection updates when the zoom threshold is passed, while the logic inside of `updateProjection` determines the correct projection. It's important to separate these two checks because if the user zooms in before changing the projection to globe, the map should render as Mercator.

The logic here could be placed in its own function, but this is the only place where it would be called.

Reply:
Refer to my comment in https://github.com/mapbox/mapbox-gl-js/pull/11460#discussion_r802121401, maybe there's room for splitting things up to reduce the scope of `_updateProjection` for a more lean and specific case between what the stylesheet wants to do when updating the projection vs what is needed for a transition.